### PR TITLE
Add optimization level 2 flag in extension Makefiles and fix warnings corresponding to maybe-uninitialized flag

### PR DESCRIPTION
**Pull request by HarshLunagariya**
### Description

`-Wmaybe-uninitialized` flag turns on with `-Wall` flag if any optimization level is enabled. This commit add optimization level 2 -O2 flag to extension makefiles and resolves error related to maybe-uninitialized flag.

### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**
NA

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by surendravishnoi — DISMISSED** [review]

**Review by sumitj824 — APPROVED** [review]

File: contrib/babelfishpg_tsql/runtime/functions.c [hunk 1]
```diff
@@ -573,7 +573,8 @@ schema_id(PG_FUNCTION_ARGS)
 
 	id = get_namespace_oid(physical_name, true);
 
-	pfree(name);
+	if (!name)
```
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/runtime/functions.c]: Shouldn't this be `if(name)` instead of `if(!name)`?

Comment (HarshLunagariya) [reply contrib/babelfishpg_tsql/runtime/functions.c]: Addressed.

**Review by rishabhtanwar29 — DISMISSED** [review]
